src/test/java/com/wikia/webdriver/common/testnglisteners/BrowserAndTestEventListener.java

```diff
@@ -50,115 +46,130 @@ public void beforeNavigateTo(String url, WebDriver driver) {
   @Override
   public void afterNavigateTo(String url, WebDriver driver) {
     Method method = TestContext.getCurrentTestMethod();
-    Class<?> declaringClass = method.getDeclaringClass();
+    if (method != null) {
+      Class<?> declaringClass = method.getDeclaringClass();
 
-    String cookieDomain = String.format(".%s", Configuration.getEnvType().getWikiaDomain());
+      String cookieDomain = String.format(".%s", Configuration.getEnvType().getWikiaDomain());
 
-    Date cookieDate = new Date(new DateTime().plusYears(10).getMillis());
+      Date cookieDate = new Date(new DateTime().plusYears(10).getMillis());
 
-    if (!AlertHandler.isAlertPresent(driver)) {
-      String command = "Url after navigation";
-      if (url.equals(driver.getCurrentUrl())) {
-        Log.ok(command, VelocityWrapper.fillLink(driver.getCurrentUrl(), driver.getCurrentUrl()));
-      } else {
-        // A fast lane to stop executing any test on "not a valid community" page
-        if(driver.getCurrentUrl().contains(URLsContent.NOT_A_VALID_COMMUNITY)){
-          throw new SkipException(String.format("Wrong redirect to: %s", driver.getCurrentUrl()));
-        }
-        if (driver.getCurrentUrl().contains("data:text/html,chromewebdata ")) {
-          driver.get(url);
-          Log.warning(command, driver.getCurrentUrl());
+      if (!AlertHandler.isAlertPresent(driver)) {
+        String command = "Url after navigation";
+        if (url.equals(driver.getCurrentUrl())) {
+          Log.ok(command, VelocityWrapper.fillLink(driver.getCurrentUrl(), driver.getCurrentUrl()));
         } else {
-          Log.warning(command, driver.getCurrentUrl());
+          // A fast lane to stop executing any test on "not a valid community" page
+          if (driver.getCurrentUrl().contains(URLsContent.NOT_A_VALID_COMMUNITY)) {
+            throw new SkipException(String.format("Wrong redirect to: %s", driver.getCurrentUrl()));
+          }
+          if (driver.getCurrentUrl().contains("data:text/html,chromewebdata ")) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS134)